app/models/agents/s3_agent.rb

```diff
@@ -233,22 +233,21 @@ def emit_event?(event_type)
     end
 
     def process_file?(file_name)
-      return true unless filename_filter_present?
-      prefix_match?(file_name) || suffix_match?(file_name)
+      prefix_match?(file_name) && suffix_match?(file_name)
     end
 
     def prefix_match?(file_name)
       prefix = interpolated['prefix']
+      return true if prefix.blank?
+
       prefix.present? && file_name.starts_with?(prefix)
```

Comment:
Wouldn't the present check here be covered by the blank check above?

Reply:
🤦‍♂️ I missed that